youtube_dl/extractor/chaturbate.py

```diff
@@ -7,7 +7,7 @@
 
 
 class ChaturbateIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:[^/]+\.)?chaturbate\.com/(?P<id>[^/?#]+)'
+    _VALID_URL = r'https?://(?:[^/]+\.)?chaturbate\.com/(?:fullvideo/\?b=)?(?P<id>[^/?&#]+)'
```

Comment:
Won't match https://chaturbate.com/fullvideo/?foo&b=... and similar.

Reply:
It actually does, because the regexp is not anchored to end of string (does not end with `$`).

But now that you mention it, I added `^` to the beginning of this regex.

Reply:
Nothing to do with my comment. Have you read it at all?

Reply:
Sorry, my bad, I misread your comment. But yes, I think that still counts as reading. Anyway it's a moot point now, thanks! :)